src/lake/Lake/CLI/Init.lean

```diff
@@ -93,13 +93,16 @@ def mathConfigFileContents (pkgName libRoot : String) :=
 s!"import Lake
 open Lake DSL
 
-def moreServerArgs := #[
-  \"-Dpp.unicode.fun=true\", -- pretty-prints `fun a ↦ b`
-  \"-Dpp.proofs.withType=false\"
+def moreServerOptions : Array ServerOption := #[
+  ⟨`pp.unicode.fun, true⟩, -- pretty-prints `fun a ↦ b`
+  ⟨`pp.proofs.withType, false⟩
 ]
 
 -- These settings only apply during `lake build`, but not in VSCode editor.
-def moreLeanArgs := moreServerArgs
+def moreLeanArgs := #[
+  \"-Dpp.unicode.fun=true\", -- pretty-prints `fun a ↦ b`
+  \"-Dpp.proofs.withType=false\"
+]
```

Comment:
@mhuisi, is there some way we can keep these in sync?

Reply:
I could provide a function to format `ServerOption` as a command line argument.

Reply:
And every newbie user is seeing this metaprogramming code in their `lake new my-first-project math` lakefile? :sob:

Reply:
I honestly don't care if we have the duplication in the template or not. Please just agree on either one!

Reply:
I don't want duplication for sure, but I also want the builtin templates to not contain any dynamic features, i.e. they are expressible as JSON with fancy syntax. In this case that could be addressed by having the defaults set up such that you don't have to specify `moreLeanArgs` at all, or there is some common `buildOptions` thing that feeds into both `moreServerOptions` and `moreLeanArgs`. Copy pasting a nontrivial program into everyone's lakefile is going to make it harder to analyze lean projects at large going forward.